### PR TITLE
SAST reusable workflow documentation and additional functionality

### PR DESCRIPTION
**Pull request by pritchyspritch**
Added documentation for GitHub Compliance App, and logic to allow semgrep in case repo is private or developer preference.

## Context
v1 of this workflow would not have worked for private repos due to licensing issues. 

## Changes proposed in this pull request
This version includes:
* checks to see if the repo is private which triggers semgrep instead of codeql
* the option to force semgrep if a team prefers it to codeql
* naming and uploading of artefacts so they can be picked up by the security teams reporting tools
* extra documentation to explain the compliance check, and how to ensure it runs in your pipeline
* the compliance job will now run so long as the repo is public, now that the app has been installed in this org and secrets are available

## Guidance to review
Successful run on test repo, shows build breaking due to security issues found outside of SLA policies: https://github.com/DFE-Digital/test-codeql/actions/runs/9909251763

## Checklist

- [X] I have performed a self-review of my code, including formatting and typos
- [X] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [X] I have added the `Devops` label
- [ ] I have attached the pull request to the trello card

### CODE REVIEW

File: .github/workflows/reusable-workflow-sast.yml [hunk 1]
```diff
@@ -75,9 +79,79 @@ on:
...
+            echo $status_code
+            visibility=public
+            echo "visibility=public" >> "$GITHUB_OUTPUT"
+          else
+            echo $status_code
+            echo "visibility=private" >> "$GITHUB_OUTPUT"
+            visibility=private
+          fi
+
+          echo ${{ github.actor }}
```
Comment (saliceti) [inline .github/workflows/reusable-workflow-sast.yml:107]: Is this debug code that can be removed?

File: .github/workflows/reusable-workflow-sast.yml [hunk 2]
```diff
@@ -75,9 +79,79 @@ on:
...
+    steps:
+      - uses: actions/checkout@v4
+
+      # Run the "semgrep scan" command on the command line of the docker image.
+      - run: semgrep scan --config auto --sarif --sarif-output=semgrep.sarif --force-color
+
+      - name: Name sarif outputs
+        id: name-sarifs
+        run: |
+          service=$(echo ${{ github.repository_owner }}_${{ github.event.repository.name }} | tr '[:upper:]' '[:lower:]' | sed -e 's|/|_|g' -e 's|\.|_|g' -e 's| |_|g' -e 's|-|_|g')
```
Comment (saliceti) [inline .github/workflows/reusable-workflow-sast.yml:138]: Can you add a comment explaining what this line does? It's a bit obscure

File: .github/workflows/reusable-workflow-sast.yml [hunk 3]
```diff
@@ -89,12 +163,19 @@ jobs:
       matrix:
         location: ${{ fromJSON(inputs.dotnet_project_locations) }}
 
+    if: ${{ (github.actor != 'dependabot[bot]') && (needs.test-visibility.outputs.visibility == 'public' && inputs.force_semgrep == false) }}
     steps:
 
+      - name: test outputs
```
Comment (saliceti) [inline .github/workflows/reusable-workflow-sast.yml:169]: Is this debug code that can be removed?

File: .github/workflows/reusable-workflow-sast.yml (hunk 1 above)
Comment (pritchyspritch) [reply .github/workflows/reusable-workflow-sast.yml]: Removed 👍🏻

File: .github/workflows/reusable-workflow-sast.yml (hunk 2 above)
Comment (pritchyspritch) [reply .github/workflows/reusable-workflow-sast.yml:137]: Comment added above the run, good point 👍🏻

File: .github/workflows/reusable-workflow-sast.yml (hunk 3 above)
Comment (pritchyspritch) [reply .github/workflows/reusable-workflow-sast.yml]: Another good spot - removed 👍🏻

**Review by saliceti — APPROVED** [review]

**Pull request merged**